Userland/Utilities/tail.cpp

```diff
@@ -77,19 +53,59 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     TRY(Core::System::pledge("stdio rpath"));
 
     bool follow = false;
-    int line_count = DEFAULT_LINE_COUNT;
-    char const* file = nullptr;
+    size_t line_count = DEFAULT_LINE_COUNT;
+    StringView file;
 
     Core::ArgsParser args_parser;
-    args_parser.set_general_help("Print the end ('tail') of a file.");
+    args_parser.set_general_help("Print the end - the \"tail\" - of a file.");
     args_parser.add_option(follow, "Output data as it is written to the file", "follow", 'f');
     args_parser.add_option(line_count, "Fetch the specified number of lines", "lines", 'n', "number");
-    args_parser.add_positional_argument(file, "File path", "file");
+    args_parser.add_positional_argument(file, "File path", "file", Core::ArgsParser::Required::No);
     args_parser.parse(arguments);
 
-    auto f = TRY(Core::File::open(file, Core::OpenMode::ReadOnly));
-    TRY(Core::System::pledge("stdio"));
+    // Since we can't seek the standard input, we read
+    // the entire file and split on new lines.
+    if (file.is_empty() || file == "-") {
+        do {
+            auto buffer = TRY(TRY(Core::Stream::File::standard_input())->read_all(PAGE_SIZE));
+            auto lines = StringView(buffer).split_view("\n"sv, true);
+            int i = 0;
```

Comment:
`i` musn't be unsigned since lines.size() - line_count might be negative. (using size_t will make the subtraction overflow, and the >= condition will thus be false)